domain/application/state/application.go

```diff
@@ -874,30 +880,168 @@ INSERT INTO k8s_service (*) VALUES ($cloudService.*)
 		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
 			return errors.Errorf(
 				"querying cloud service for application %q and provider id %q: %w", applicationName, providerID, err)
+		} else if errors.Is(err, sqlair.ErrNoRows) {
+			// Nothing already exists so create a new net node and placeholder
+			// link layer device for the cloud service.
+			nodeUUID, err := st.insertCloudServiceNetNode(ctx, tx)
+			if err != nil {
+				return errors.Errorf("creating cloud service net node for application %q: %w", applicationName, err)
+			}
+			serviceInfo.NetNodeUUID = nodeUUID.String()
+
+			uuid, err := uuid.NewUUID()
+			if err != nil {
+				return errors.Capture(err)
+			}
+			serviceInfo.UUID = uuid.String()
+			if err := tx.Query(ctx, insertStmt, serviceInfo).Run(); err != nil {
+				return errors.Capture(err)
+			}
```

Comment:
Can we break this out into a separate method

Reply:
Yes absolutely. Fixed.